superset/security/manager.py

```diff
@@ -974,14 +976,14 @@ def set_perm(  # pylint: disable=unused-argument
                 )
 
     def raise_for_access(
-        # pylint: disable=too-many-arguments,too-many-locals
+        # pylint: disable=too-many-arguments
         self,
-        database: Optional["Database"] = None,
-        datasource: Optional["BaseDatasource"] = None,
-        query: Optional["Query"] = None,
-        query_context: Optional["QueryContext"] = None,
-        table: Optional["Table"] = None,
-        viz: Optional["BaseViz"] = None,
+        database: Optional[Database] = None,
+        datasource: Optional[BaseDatasource] = None,
+        query: Optional[Query] = None,
+        query_context: Optional[QueryContext] = None,
+        table: Optional[Table] = None,
+        viz: Optional[BaseViz] = None,
```

Comment:
Is this nowadays the correct way of referring to imports from `TYPE_CHECKING`?

Reply:
a string representation is worst than the real type

Reply:
If the imports are inside an `if TYPE_CHECKING` then you need to use a string, otherwise you get a `NameError`.

Reply:
> 
When you use the old method of type hint as string, you actually don't need to import the class at all and don't need to put it under TYPE_CHECKING
but if you would like to use the new better choice for marking the type hint by the real type 
you should import them under TYPE_CHECKING with the __future__ annotations.
That is the better choice so you can navigate and let the IDEs do their job --> assist you